internal/build.go

```diff
@@ -64,6 +79,10 @@ func RunBuild(args []string, params *BuildCommandParams) error {
 		return err
 	}
 
+	if metadataFile != "" {
+		_ = os.Remove(metadataFile)
+	}
+
 	return out.Close()
```

Comment:
Both this `os.Remove()` call and the `out.Close()` I think would be better served to live in a `defer` block, that way we ensure they're run regardless of whether this function returns early or not.